### ISSUE TITLE
Wrong variant returned for `invalid_api_key_indexes` error when creating an API key.

### ISSUE THREAD

**Issue by gmourier** [meilisearch/meilisearch#2924]
> Sending a value that is not a valid index (e.g. regarding the index uid format) in the indexes field returns an [invalid_api_key_indexes](https://github.com/meilisearch/meilisearch/issues/0061-error-format-and-definitions.md#invalid_api_key_indexes) with a variant error message. See "Sending an invalid index uid format in indexes field" variant.

It should have been solved by https://github.com/meilisearch/meilisearch/issues/2158, but I did not get the new variant testing on the latest version.

```json
{
    "message": "`indexes` field value `[\"invalid index # / \\name with spaces\"]` is invalid. It should be an array of string representing index names.",
    "code": "invalid_api_key_indexes",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#invalid_api_key_indexes"
}
```

while I should receive

```json
{
    "message": "` [\"invalid index # / \\name with spaces\"]` is not a valid index uid. Index uid can be an integer or a string containing only alphanumeric characters, hyphens (-) and underscores (_).",
    "code": "invalid_api_key_indexes",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#invalid_api_key_indexes"
}
```

Relates to that catch-up specification https://github.com/meilisearch/specifications/pull/181 (The original one was never merged 🤔 So it makes me doubt if it was solved)

**Comment by gmourier** [comment]
Hi everyone 👋

I just tested on `main` and checked the code, the used variant still doesn't seem to be the right one.

I'm still receiving 

```json
{
    "message": "`[\"&@----invalid\"]` is not a valid index uid. It should be an array of string representing index names.",
    "code": "invalid_api_key_indexes",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#invalid_api_key_indexes"
}
```

while the spec mentions

```json
{
    "message": "`[\"&@----invalid\"]` is not a valid index uid. Index uid can be an integer or a string containing only alphanumeric characters, hyphens (-) and underscores (_).",
    "code": "invalid_api_key_indexes",
    "type": "invalid_request",
    "link": "https://docs.meilisearch.com/errors#invalid_api_key_indexes"
}
```

1. If the `indexes` array contains something different than a string, the returned variant is the right one
2. If the `indexes` array contains strings that aren't valid regarding an indexUid format, the provided variant by the specification is expected.

**Comment by curquiza** [comment]
Closed by https://github.com/meilisearch/meilisearch/pull/2930

**Comment by gmourier** [comment]
Hello 👋 

By looking at the code and testing it, the wrong variant is returned.

**Comment by ManyTheFish** [comment]
Fixed by #3032
